From 61443632dc5ea69a072612b6fa3f7da62c96b72c Mon Sep 17 00:00:00 2001
From: Kevin Tan <62129747+kevintsq@users.noreply.github.com>
Date: Mon, 13 Mar 2023 17:48:57 +0800
Subject: [PATCH] Modify import to support librosa==0.10.0

---
 piano_transcription_inference/utilities.py | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/piano_transcription_inference/utilities.py b/piano_transcription_inference/utilities.py
index e73e301..8040c45 100644
--- a/piano_transcription_inference/utilities.py
+++ b/piano_transcription_inference/utilities.py
@@ -1,7 +1,7 @@
 import os
 import numpy as np
 import audioread
-import librosa
+from librosa.core import audio
 from mido import MidiFile
 
 from .piano_vad import (note_detection_with_onset_offset_regress, 
@@ -520,7 +520,7 @@ def load_audio(path, sr=22050, mono=True, offset=0.0, duration=None,
         n = 0
 
         for frame in input_file:
-            frame = librosa.core.audio.util.buf_to_float(frame, dtype=dtype)
+            frame = audio.util.buf_to_float(frame, dtype=dtype)
             n_prev = n
             n = n + len(frame)
 
@@ -550,10 +550,10 @@ def load_audio(path, sr=22050, mono=True, offset=0.0, duration=None,
         if n_channels > 1:
             y = y.reshape((-1, n_channels)).T
             if mono:
-                y = librosa.core.audio.to_mono(y)
+                y = audio.to_mono(y)
 
         if sr is not None:
-            y = librosa.core.audio.resample(y, orig_sr=sr_native, target_sr=sr, res_type=res_type)
+            y = audio.resample(y, orig_sr=sr_native, target_sr=sr, res_type=res_type)
 
         else:
             sr = sr_native
@@ -561,4 +561,4 @@ def load_audio(path, sr=22050, mono=True, offset=0.0, duration=None,
     # Final cleanup for dtype and contiguity
     y = np.ascontiguousarray(y, dtype=dtype)
 
-    return (y, sr)
\ No newline at end of file
+    return (y, sr)
